specification/hybridcompute/resource-manager/Microsoft.HybridCompute/stable/2022-11-10/HybridCompute.json

```diff
@@ -0,0 +1,1675 @@
+            "description": "The name of the machine containing the extension."
+          },
+          {
+            "name": "extensionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the machine extension."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/MachineExtension"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "GET Machine Extension": {
+            "$ref": "./examples/GETExtension.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HybridCompute/machines/{machineName}/extensions": {
+      "get": {
+        "tags": [
+          "MachineExtensions"
+        ],
+        "operationId": "MachineExtensions_List",
+        "description": "The operation to get all extensions of a non-Azure machine",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "machineName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the machine containing the extension."
+          },
+          {
+            "name": "$expand",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "The expand expression to apply on the operation."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/MachineExtensionsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
```

Comment:
Please replace all `v2` common type refs with `v3` for consistentcy.

Reply:
@timotheeguerin this swagger uses a mixed v2 and v3 common types of `ErrorResponse` and `ErrorDetails`. Their definitions are exactly the same in both versions, but modelerfour gave the below errors when generating .NET SDK:

>  error   | PreCheck/DuplicateSchema | Duplicate Schema named 'ErrorDetail' (1 differences):
>     - properties.details.$ref: "#/components/schemas/schemas:258" => "#/components/schemas/schemas:356"; 
>  error   | PreCheck/DuplicateSchema | Duplicate Schema named 'ErrorResponse' (1 differences):
>     - properties.error.$ref: "#/components/schemas/schemas:254" => "#/components/schemas/schemas:352"; 

Is this expected?

Reply:
yes, modelerfour doesn't support deduplicating types with circular references

Reply:
it's fixed